diff --git a/Cargo.toml b/Cargo.toml
index b55e329..9aa3723 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -2,7 +2,6 @@
 name = "gopher64"
 version = "1.1.17"
 edition = "2024"
-rust-version = "1.95.0"

 [features]
 default = ["gui"]
@@ -60,4 +59,4 @@ identifier = "io.github.gopher64.gopher64"
 category = "public.app-category.games"
 icon = ["data/icon/gopher64_128x128.png","data/icon/gopher64_256x256.png","data/icon/gopher64_512x512.png"]
 osx_frameworks = ["/opt/homebrew/opt/molten-vk/lib/libMoltenVK.dylib","/opt/homebrew/opt/freetype/lib/libfreetype.6.dylib","/opt/homebrew/opt/libpng/lib/libpng16.16.dylib"]
-osx_minimum_system_version = "15.7"
+osx_minimum_system_version = "15.0"
diff --git a/rust-toolchain.toml b/rust-toolchain.toml
deleted file mode 100644
index f25b5b1..0000000
--- a/rust-toolchain.toml
+++ /dev/null
@@ -1,2 +0,0 @@
-[toolchain]
-channel = "1.95.0"

