From 79562db355cfedba14af84aa204f16015dbbb403 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 05 Feb 2023 09:36:17 +0000
Subject: [PATCH] path changes

diff --git a/vendor/go.podman.io/common/pkg/config/config_linux.go b/vendor/go.podman.io/common/pkg/config/config_linux.go
index e0b4672e7..f9e68d8ac 100644
--- a/vendor/go.podman.io/common/pkg/config/config_linux.go
+++ b/vendor/go.podman.io/common/pkg/config/config_linux.go
@@ -5,10 +5,8 @@ import (
 )
 
 var defaultHelperBinariesDir = []string{
-	"/usr/local/libexec/podman",
-	"/usr/local/lib/podman",
-	"/usr/libexec/podman",
-	"/usr/lib/podman",
+	"/usr/libexec/podman",
+	"/usr/bin",
 }
 
 // Capabilities returns the capabilities parses the Add and Drop capability
-- 
2.53.0

