diff --git a/src/steamcompmgr.cpp b/src/steamcompmgr.cpp
--- a/src/steamcompmgr.cpp
+++ b/src/steamcompmgr.cpp
@@ -2020,23 +2020,30 @@
 
 void MouseCursor::UpdatePosition()
 {
+	Window root_ret, child_ret;
+	int root_x = m_x, root_y = m_y, win_x = 0, win_y = 0;
+	unsigned int mask;
+	XQueryPointer( m_ctx->dpy, DefaultRootWindow( m_ctx->dpy ),
+	               &root_ret, &child_ret,
+	               &root_x, &root_y,
+	               &win_x, &win_y,
+	               &mask );
+
 	wlserver_lock();
-	struct wlr_pointer_constraint_v1 *pConstraint = wlserver.GetCursorConstraint();
-	if ( pConstraint && pConstraint->current.cursor_hint.enabled )
-	{
-		m_bConstrained = true;
-
-		m_x = pConstraint->current.cursor_hint.x;
-		m_y = pConstraint->current.cursor_hint.y;
-	}
-	else
-	{
-		m_bConstrained = false;
-
-		m_x = wlserver.mouse_surface_cursorx;
-		m_y = wlserver.mouse_surface_cursory;
-	}
+	m_bConstrained = !!wlserver.GetCursorConstraint();
 	wlserver_unlock();
+
+	if ( root_x != m_x || root_y != m_y )
+	{
+		m_x = root_x;
+		m_y = root_y;
+		if ( !m_imageEmpty )
+		{
+			wlserver.ulLastMovedCursorTime = get_time_in_nanos();
+			wlserver.bCursorHidden = !wlserver.bCursorHasImage;
+			hasRepaintNonBasePlane = true;
+		}
+	}
 }
 
 void MouseCursor::checkSuspension()
