diff --git a/src/crypto/x509/root_linux.go b/src/crypto/x509/root_linux.go
index ad6ce5cae7..763c686fed 100644
--- a/src/crypto/x509/root_linux.go
+++ b/src/crypto/x509/root_linux.go
@@ -19,6 +19,7 @@ var certFiles = []string{
 var certDirectories = []string{
 	"/etc/ssl/certs",     // SLES10/SLES11, https://golang.org/issue/12139
 	"/etc/pki/tls/certs", // Fedora/RHEL
+	"/etc/ssl",           // ROCKNIX
 }
 
 func init() {
